sgkit/stats/ld.py

```diff
@@ -61,3 +70,377 @@ def rogers_huff_r_between(gn0: ArrayLike, gn1: ArrayLike) -> float:  # pragma: n
 @njit(nogil=True, fastmath=True, cache=True)  # type: ignore
 def rogers_huff_r2_between(gn0: ArrayLike, gn1: ArrayLike) -> float:  # pragma: no cover
     return rogers_huff_r_between(gn0, gn1) ** 2  # type: ignore
+
+
+def ld_matrix(
+    ds: Dataset,
+    *,
+    dosage: Hashable = variables.dosage,
+    threshold: Optional[float] = None,
+    ld_score: Optional[Hashable] = None,
+) -> DataFrame:
+    """Compute a sparse linkage disequilibrium (LD) matrix.
+
+    This method computes the Rogers Huff R2 value for each pair of variants in
+    a window, and returns those that exceed the provided threshold, as a sparse
+    matrix dataframe.
+
+    Parameters
+    ----------
+    ds
+        Dataset containing genotype dosages.
+    dosage
+        Name of genetic dosage variable.
+        Defined by :data:`sgkit.variables.dosage_spec`.
+    threshold
+        R2 threshold below which no variant pairs will be returned. This should almost
+        always be something at least slightly above 0 to avoid the large density very
+        near zero LD present in most datasets.
+    ld_score
+        Optional name of variable to use to prioritize variant selection
+        (e.g. minor allele frequency). Defaults to None.
+        Defined by :data:`sgkit.variables.ld_score_spec`.
+
+    Returns
+    -------
+    Upper triangle (including diagonal) of LD matrix as COO in dataframe.  Fields:
+
+    - ``i``: Row (variant) index 1
+    - ``j``: Row (variant) index 2
+    - ``value``: R2 value
+    - ``cmp``: If ``ld_score`` is provided, this is 1, 0, or -1 indicating whether or not ``i > j`` (1), ``i < j`` (-1), or ``i == j`` (0)
+
+    Raises
+    ------
+    ValueError
+        If the dataset is not windowed.
+    """
+
+    if not has_windows(ds):
+        raise ValueError("Dataset must be windowed for ld_matrix")
```

Comment:
Mention the `sgkit.window` function so users know what to do next?

Reply:
Done